ci-operator/step-registry/fips-check/optional-operator-images-scan/fips-check-optional-operator-images-scan-commands.sh

```diff
@@ -90,81 +90,83 @@ mkdir $data_dir || true
         fi
+        # perform fips scan by using check-payload tool
         for image_url in ${image_list}; do
-            imges_test_result=$(ssh -o UserKnownHostsFile=/dev/null -o IdentityFile="${SSH_PRIV_KEY_PATH}" -o StrictHostKeyChecking=no ${BASTION_SSH_USER}@${BASTION_IP} \
-                "sudo podman run --privileged -it -v /tmp/tmp/:/root/:ro registry.ci.openshift.org/ci/check-payload scan operator --spec $image_url" || true)
-            echo "###test result for $image_url is: $imges_test_result" >> $images_test_result
+            image_test_result=$(ssh -o UserKnownHostsFile=/dev/null -o IdentityFile="${SSH_PRIV_KEY_PATH}" -o StrictHostKeyChecking=no ${BASTION_SSH_USER}@${BASTION_IP} 2> /dev/null \
+                "sudo podman run --rm --privileged -v /tmp/tmp/:/root/:ro registry.ci.openshift.org/ci/check-payload scan operator --spec $image_url" || true)
+            echo "###test result for image '$image_url' is: $image_test_result" >> $csv_test_result
         done
-        echo "cat ${imges_test_result}" >> ${scan_result_file}
-        resFail=$(grep -E "Failure Report" ${images_test_result} || true)
-        resSuessWithWarnings=$(grep -E "Successful run with warnings" ${images_test_result} || true)
+        echo "$(cat ${csv_test_result})" >> ${scan_result_file}
+        # summarize scan results from csv_test_result file
+        resFail=$(grep -E "Failure Report" ${csv_test_result} || true)
+        resSuessWithWarnings=$(grep -E "Successful run with warnings" ${csv_test_result} || true)
         if [[ -n $resFail ]];then
-            echo "Fips check  for csv $csv in package $package failed!"
+            echo "##Fips check for csv '$csv' in package '$package' failed!"
             echo "$csv" >> ${scan_result_failure_csvs}
             pass=false
         elif [[ -n $resSuessWithWarnings ]];then
-            echo "Fips check  for csv $csv in package $package succeed with warnings!"
+            echo "##Fips check for csv '$csv' in package '$package' succeed with warnings!"
             echo "$csv" >> ${scan_result_succeed_with_warnings_csvs}
             pass=false
         else
-            echo "Fips check for csv $csv in package $package succeed!"
+            echo "##Fips check for csv '$csv' in package '$package' succeed!"
+            echo "$csv" >> ${scan_result_succeed_csvs}
         fi
-        #delete containers and images to save disk space
-        echo "delete containers"
-        ssh -o UserKnownHostsFile=/dev/null -o IdentityFile="${SSH_PRIV_KEY_PATH}" -o StrictHostKeyChecking=no ${BASTION_SSH_USER}@${BASTION_IP} \
-            "for container in \$(sudo podman ps -q -a); do sudo podman rm \${container} || true; done"
-        echo "delete images"
-        ssh -o UserKnownHostsFile=/dev/null -o IdentityFile="${SSH_PRIV_KEY_PATH}" -o StrictHostKeyChecking=no ${BASTION_SSH_USER}@${BASTION_IP} \
-            "podman rmi $(echo ${image_list} | sed 's/\"//g') || true"
```

Comment:
No need here. Because each check-payload scan will be executed in a container. That means all related images pulled will only exist within that container. Simply removing the container (by adding `--rm`) will be sufficient

https://github.com/openshift/release/blob/924faeed0f7a16874303f9a7ef1c2e30b0347174/ci-operator/step-registry/fips-check/optional-operator-images-scan/fips-check-optional-operator-images-scan-commands.sh#L143

Reply:
I also got the chance to see which images will be there (on the host) when debugging:
```
[debug] list image to see how many exist before prune
REPOSITORY TAG IMAGE ID CREATED SIZE
registry.ci.openshift.org/ci/check-payload latest 6ca6edf050b1 10 hours ago 1.59 GB
quay.io/crcont/squid latest 412d448cd802 3 years ago 302 MB
quay.io/openshifttest/registry 2 2d4f4b5309b1 4 years ago 26.8 MB
```

The above images will be used throughout the scan process, so they do not need to be deleted.